File: app/src/main/java/wyyu/multi/normal/ActivityListNormal.java
package wyyu.multi.normal;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import java.util.ArrayList;
import java.util.List;
import rx.Observable;
import rx.Subscriber;
import rx.android.schedulers.AndroidSchedulers;
import rx.functions.Action1;
import rx.schedulers.Schedulers;
import wyyu.multi.R;
import wyyu.multi.app.Constants;
import wyyu.multi.normal.adapter.AdapterListNormal;
import wyyu.multi.normal.data.DataNormal;

/**
 * Created by wyyu on 2020-05-07.
 **/

public class ActivityListNormal extends AppCompatActivity {

    public static void open(Context context) {
        context.startActivity(new Intent(context, ActivityListNormal.class));
    }

    @BindView(R.id.normal_recycler) RecyclerView recyclerView;

    private AdapterListNormal adapter;
    private Unbinder unbinder;

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_normal);

        unbinder = ButterKnife.bind(this);
        initActivity();
    }

    @Override protected void onDestroy() {
        super.onDestroy();
        if (unbinder != null) {
            unbinder.unbind();
        }
    }

    private void initActivity() {
        initList();
        loadData();
    }

    private void initList() {
        adapter = new AdapterListNormal();

        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setItemAnimator(null);
        recyclerView.setAdapter(adapter);
    }

    private void loadData() {
        Observable.unsafeCreate(new Observable.OnSubscribe<List<DataNormal>>() {
            @Override public void call(Subscriber<? super List<DataNormal>> subscriber) {
                List<DataNormal> dataList = new ArrayList<>();
                for (int index = 0; index < Constants.COUNT; index++) {
                    dataList.add(new DataNormal(index));
                }
                subscriber.onNext(dataList);
                subscriber.onCompleted();
            }
        })
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe(new Action1<List<DataNormal>>() {
                @Override public void call(List<DataNormal> dataList) {
                    if (adapter != null) {
                        adapter.setDataList(dataList);
                    }
                }
            }, new Action1<Throwable>() {
                @Override public void call(Throwable throwable) {
                    Log.e("", throwable.getMessage());
                }
            });
    }
}

File: app/src/main/java/com/wyyu/multiktest/normal/ActivityListNormal.kt
package com.wyyu.multiktest.normal

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.os.PersistableBundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.wyyu.multiktest.app.Constants
import com.wyyu.multiktest.databinding.ActivityListNormalBinding
import com.wyyu.multiktest.normal.adapter.ListNormalAdapter
import com.wyyu.multiktest.normal.data.DataNormal

/**
 * Created by wyyu on 2021/7/30.
 **/

class ActivityListNormal : AppCompatActivity() {

    companion object {
        fun open(context: Context) {
            context.startActivity(Intent(context, ActivityListNormal::class.java))
        }
    }

    private var binding: ActivityListNormalBinding? = null

    private var adapter: ListNormalAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityListNormalBinding.inflate(layoutInflater)
        setContentView(binding?.root)

        initActivity()
    }

    private fun initActivity() {
        initList()
        loadList()
    }

    private fun initList() {
        adapter = ListNormalAdapter()

        binding?.listNormalRecycler?.adapter = adapter
        binding?.listNormalRecycler?.layoutManager = LinearLayoutManager(this)
    }

    private fun loadList() {
        val list = ArrayList<DataNormal>()
        for (index in 0 until Constants.COUNT) {
            list.add(DataNormal(index))
        }
        adapter?.initList(list)
    }
}